### ISSUE TITLE
Performance drops significantly whenever hands are visible due to WindowsMixedRealityXRSDKArticulatedHand.

### ISSUE THREAD

**Issue by emctague** [microsoft/MixedRealityToolkit-Unity#9510]
## Describe the bug

When hands are visible in the HoloLens 2's field of vision, `WindowsMixedRealityXRSDKArticulatedHand.UpdateController` profiles as taking *as long as 60 milliseconds* to execute per frame.

## To reproduce

1. Create a cookie-cutter HoloLens 2 project.
2. Begin profiling in Unity.
3. Hold your hands up to your face and watch the framerate drop to nothing!

## Expected behavior

The existence of hands should not make my app lag.

## Screenshots

![image](https://user-images.githubusercontent.com/22970065/111342996-5d41d300-8651-11eb-97c4-e33bf9387f9e.png)
![image](https://user-images.githubusercontent.com/22970065/111343053-6d59b280-8651-11eb-86e1-35f5cb98ec88.png)

## Your setup (please complete the following information)

- Unity Version 2020.3.0f1
- MRTK Version 2.5.4

## Target platform (please complete the following information)

- HoloLens 2

## Additional context

This is essentially a *default* project. The entire HoloLens 2 doesn't lag struggling to track my hands outside of this project, so I don't imagine it should be too expensive for it to do the same here.

**Comment by keveleigh** [comment]
Are you profiling on-device or through holographic remoting?

**Comment by emctague** [comment]
I am profiling on-device. Took a second look through project issues and this seems to be the same issue as #9400. Important to note that two visible hands causes *double* the lag, and that mesh visualization is turned off entirely. Perhaps if I set the Joint/Palm/Fintertip/Hand Mesh prefabs to None I may see a performance difference, as those smallest green actions appear to all be related to transforming the hand mesh.

**Comment by polar-kev** [comment]
hey @emctague, are you also seeing this on MRTK 2.6.1?

**Comment by emctague** [comment]
@polar-kev Just upgraded, still experiencing the issue exactly as described.

**Comment by praptoumache** [comment]
> @polar-kev Just upgraded, still experiencing the issue exactly as described.
> 
> **Edit:** My bad, the issue seems to be resolved. Thank you!

having the exact same issue. how did you resolve the issue?

**Comment by rfurmaniak** [comment]
Experiencing the same issue with MRTK 2.8.2.
Comparison in the profiler between hands visible and hands invisible (higher frame time chart indicates that hands are visible):
![image](https://user-images.githubusercontent.com/50983064/190112835-95454627-6036-452b-a8a6-fd770ec8f91d.png)

Deep profile with hands off:
![image](https://user-images.githubusercontent.com/50983064/190112868-43f9c29f-35ad-4b20-9498-82bb277ceb04.png)

Deep profile with hands on:
![image](https://user-images.githubusercontent.com/50983064/190112911-9036d027-014a-4d04-ad76-a49c6ea76e37.png)

**Comment by Schroedingers-Cat** [comment]
@rfurmaniak were you ever able to fix this problem? I've noticed this through v2.8.1-2.8.3 myself. As soon as one hand is recognized by the HL the MRTK's UpdateAllServices starts using around 2-3ms (profiled on the device without deep profiling). Any hand visualization is off. Some frames around the initial detection of the hand, the MRTK UpdateAllServices calls peak around 7ms.

@emctague the issue was resolved for you with 2.6.1 - did you update to any of the later MRTK 2.x versions and experience the problem again?
